File: 553.optimal-division.py
#
# @lc app=leetcode id=553 lang=python3
#
# [553] Optimal Division
#

# @lc code=start
class Solution:
    def optimalDivision(self, nums: List[int]) -> str:
        width = len(nums)
        if width == 1:
            return str(nums[0])
        elif width == 2:
            return str(nums[0])+'/'+str(nums[1])
        res = []
        res.append(str(nums[0]))
        res.append("/(")
        for i in range(1, width-1):
            res.append(str(nums[i]))
            res.append('/')
        res.append(str(nums[width-1]))
        res.append(')')
        return ''.join(res)

# @lc code=end

File: 553.optimal-division.cpp
/*
 * @lc app=leetcode id=553 lang=cpp
 *
 * [553] Optimal Division
 */
#include <vector>
#include <string>
using namespace std;
// @lc code=start
class Solution
{
public:
    string optimalDivision(vector<int> &nums)
    {
        int width = nums.size();
        string res;
        res.append(to_string(nums[0]));
        if (width == 1)
        {
            return res;
        }
        else if (width == 2)
        {
            res.push_back('/');
            res.append(to_string(nums[1]));
            return res;
        }
        else
        {
            res.append("/(");
        }
        for (int i = 1; i != width - 1; ++i)
        {
            res.append(to_string(nums[i]));
            res.push_back('/');
        }
        res.append(to_string(nums[width - 1]));
        res.push_back(')');
        return res;
    }
};
// @lc code=end
